pyanaconda/modules/payloads/source/factory.py

```diff
@@ -57,3 +57,25 @@ def create_source(source_type: SourceType):
             return HardDriveSourceModule()
 
         raise ValueError("Unknown source type: {}".format(source_type))
+
+    @classmethod
+    def get_type_for_kickstart(cls, data):
+        """Get a payload source type for the given kickstart data.
+
+        :param data: a kickstart data
+        :return: a payload type
```

Comment:
"a payload source type" ?

Reply:
The solution changed during your review :).